test/functional/http/post/3.votes.js

```diff
@@ -0,0 +1,437 @@
+				node.expect(res).to.have.property('message').to.equal('Invalid vote at index 0 - Invalid vote type');
+				badTransactions.push(transaction);
+			});
+		});
+
+		it('upvoting with no funds should fail', function () {
+			accountNoFunds = node.randomAccount();
+			transaction = node.lisk.vote.createVote(accountNoFunds.password, ['+' + node.eAccount.publicKey]);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.not.be.ok;
+				node.expect(res).to.have.property('message').to.equal('Account does not have enough LSK: ' + accountNoFunds.address + ' balance: 0');
+				badTransactions.push(transaction);
+			});
+		});
+
+		it('upvoting with minimal required amount of funds should be ok', function () {
+			transaction = node.lisk.vote.createVote(accountMinimalFunds.password, ['+' + node.eAccount.publicKey]);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+				goodTransactions.push(transaction);
+			});
+		});
+
+		it('downvoting not voted delegate should fail', function () {
+			transaction = node.lisk.vote.createVote(account.password, ['-' + node.eAccount.publicKey]);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.not.ok;
+				node.expect(res).to.have.property('message').to.equal('Failed to remove vote, account has not voted for this delegate');
+				badTransactions.push(transaction);
+			});
+		});
+
+		it('upvoting with valid params should be ok', function () {
+			transaction = node.lisk.vote.createVote(account.password, ['+' + node.eAccount.publicKey]);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+				goodTransactions.push(transaction);
+			});
+		});
+
+		it('self upvoting with valid params should be ok', function () {
+			transaction = node.lisk.vote.createVote(node.eAccount.password, ['+' + node.eAccount.publicKey]);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+				goodTransactions.push(transaction);
+			});
+		});
+
+		it('upvoting 33 delegates at once should be ok', function () {
+			transaction = node.lisk.vote.createVote(account33.password, delegates33.map(function (delegate) {
+				return '+' + delegate.publicKey;
+			}));
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+				goodTransactions.push(transaction);
+			});
+		});
+
```

Comment:
Why is downvoting done in a different place?

Reply:
Because It performs the downvoting over the same fresh accounts just voted in the prior phase.